pkg/wallet/wallet.go

```diff
@@ -146,7 +146,21 @@ func (w *Wallet) Save() error {
 	if err := w.rewind(); err != nil {
 		return err
 	}
-	return json.NewEncoder(w.rw).Encode(w)
+	err := json.NewEncoder(w.rw).Encode(w)
+	if err != nil {
+		return err
+	}
+
+	if f, ok := w.rw.(*os.File); ok {
```

Comment:
Move it up and `f.Truncate(0)`, then write?

Reply:
I think it is better to truncate after, in case there will be some error during encode.

Reply:
If we're to suppose encoding error we can end up with garbage written anyway (overwriting real data), so the safest thing to do would be encoding to buffer and flushing from it.